orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/StartAwsCodeBuildTask.java

```diff
@@ -51,4 +69,74 @@ public TaskResult execute(@Nonnull Stage stage) {
     context.put("buildInfo", execution);
     return TaskResult.builder(ExecutionStatus.SUCCEEDED).context(context).build();
   }
+
+  private void appendProjectName(Map<String, Object> requestInput, String projectName) {
+    requestInput.put(PROJECT_NAME, projectName);
+  }
+
+  private void appendSource(Map<String, Object> requestInput, Artifact artifact) {
+    switch (artifact.getType()) {
+      case "s3/object":
+        requestInput.put(SOURCE_TYPE, "S3");
+        String s3Location = artifact.getReference();
+        requestInput.put(
+            SOURCE_LOCATION, s3Location.startsWith("s3://") ? s3Location.substring(5) : s3Location);
+        break;
+      case "git/repo":
+        requestInput.put(SOURCE_LOCATION, artifact.getReference());
```

Comment:
I notice that the widget in deck has options for selecting branch and checkout sub-path.  Are those artifact properties ignored here?  Should they be used, or should this throw an error if they are filled out?

Reply:
Branch goes to version field and subpath is ignored as CodeBuild doesn't support subpath checkout.

Reply:
If it's not used, I think validation should be done to ensure it's not set.

Reply:
will add